plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/ScaleIOStorageAdaptor.java

```diff
@@ -214,12 +214,12 @@ public boolean disconnectPhysicalDisk(String volumePath, KVMStoragePool pool) {
 
     @Override
     public boolean disconnectPhysicalDisk(Map<String, String> volumeToDisconnect) {
-        return true;
+        return false;
     }
 
     @Override
     public boolean disconnectPhysicalDiskByPath(String localPath) {
```

Comment:
The overridden adaptor methods not processing / disconnecting the file in the local path should return '`false`' as per the comment here.

https://github.com/apache/cloudstack/blob/ac794a0c850480bd75837b003bd065b9a3d6e629/plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/StorageAdaptor.java#L53-L55

The first adaptor which returns _true_ is considered to be the adaptor that processed the volume disconnection request.

Reply:
sounds correct. I am assuming you have tested the behaviour.